tools/check-cppcheck.sh

```diff
@@ -42,5 +42,6 @@ cppcheck -j$CPPCHECK_JOBS --force \
   --error-exitcode=1 \
   --exitcode-suppressions=tools/cppcheck/suppressions-list \
   --suppressions-list=tools/cppcheck/suppressions-list \
+  -UREGEXP_RECURSION_LIMIT \
```

Comment:
This is highly suspicious. Why is this needed?

Reply:
The cppcheck print this informal message : 
```
jerry-core/ecma/operations/ecma-regexp-object.c:1: information(ConfigurationNotChecked): Skipping configuration 'REGEXP_RECURSION_LIMIT' since the value of 'REGEXP_RECURSION_LIMIT' is unknown. Use -D if you want to check it. You can use -U to skip it explicitly.
```
I will add this `ConfigurationNotChecked` to the `suppressions-list`, this supposed to be a better solution than this one.